File: androidApp/src/main/java/com/medhelp/medhelp/data/model/AllDoctorsResponse.java
//package com.medhelp.medhelp.data.model;
//
//import com.google.gson.annotations.SerializedName;
//import com.medhelp.medhelp.utils.timber_log.LoggingTree;
//
//import java.util.ArrayList;
//import java.util.List;
//
//import timber.log.Timber;
//
//public class AllDoctorsResponse {
//    @SerializedName("id_doctor")
//    private int id_doctor;
//
//    @SerializedName("full_name")
//    private String fio_doctor;
//
//    @SerializedName("id_spec")
//    private String id_specialties_string;
//
//    @SerializedName("stag")
//    private String experience;
//
//    @SerializedName("specialty")
//    private String name_specialties;
//
//    @SerializedName("dop_info")
//    private String dop_info;
//
//    @SerializedName("image_url")
//    private String image_url;
//
//    private List<Integer> id_specialties_int_list;
//
//    public List<Integer> getId_specialties_int_list() {
//        if(id_specialties_int_list==null)
//        {
//            setId_specialties_string(id_specialties_string);
//        }
//
//        return id_specialties_int_list;
//    }
//
//    public void setId_specialties_int_list(List<Integer> id_specialties_int_list) {
//        this.id_specialties_int_list = id_specialties_int_list;
//    }
//
//
//    public int getId_doctor() {
//        return id_doctor;
//    }
//
//    public void setId_doctor(int id_doctor) {
//        this.id_doctor = id_doctor;
//    }
//
//    public String getFio_doctor() {
//        return fio_doctor;
//    }
//
//    public void setFio_doctor(String fio_doctor) {
//        this.fio_doctor = fio_doctor;
//    }
//
//    public String getId_specialties_string() {
//        return id_specialties_string;
//    }
//
//    public void setId_specialties_string(String id_specialties_string) {
//        this.id_specialties_string = id_specialties_string;
//
//        if(this.id_specialties_string.isEmpty())
//            return;
//
//        id_specialties_int_list=new ArrayList<>();
//        int numberCommas= getNumberOfCommas(this.id_specialties_string)+1;
//
//        if(numberCommas==1)
//        {
//            id_specialties_int_list.add(Integer.parseInt(this.id_specialties_string));
//            return;
//        }
//
//        String str=this.id_specialties_string;
//        for(int i=0;i<numberCommas;i++)
//        {
//            int num=str.indexOf(",");
//
//            if(num!=-1) {
//                String spec = str.substring(0,num);
//                id_specialties_int_list.add(Integer.parseInt(spec));
//                str=str.substring(num+1,str.length());
//            }
//            else
//            {
//                try {
//                    id_specialties_int_list.add(Integer.parseInt(str));
//                }catch (Exception e){
//                    Timber.tag("my").e(LoggingTree.getMessageForError(e, "AllDoctorsResponse/setId_specialties_string"));
//                }
//                break;
//            }
//        }
//    }
//
//    private int getNumberOfCommas(String str)
//    {
//        int commas=0;
//        for(char ch:str.toCharArray())
//        {
//            if(ch==',')
//                commas++;
//        }
//        return commas;
//    }
//
//
//    public String getExperience() {
//        return experience;
//    }
//
//    public void setExperience(String experience) {
//        this.experience = experience;
//    }
//
//    public String getName_specialties() {
//        return name_specialties;
//    }
//
//    public void setName_specialties(String name_specialties) {
//        this.name_specialties = name_specialties;
//    }
//
//    public String getDop_info() {
//        return dop_info;
//    }
//
//    public void setDop_info(String dop_info) {
//        this.dop_info = dop_info;
//    }
//
//    public String getImage_url() {
//        return image_url;
//    }
//
//    public void setImage_url(String image_url) {
//        this.image_url = image_url;
//    }
//
//}

File: shared/src/commonMain/kotlin/com/medhelp/newmedhelp/model/AllDoctorsResponse.kt
package com.medhelp.newmedhelp.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
open class AllDoctorsResponse {
    @SerialName("id_doctor") var id = 0
    @SerialName("full_name") var fio_doctor: String? = null
    @SerialName("id_spec") var id_specialties_string: String? = null
    @SerialName("stag") var experience: String? = null
    @SerialName("specialty") var name_specialties: String? = null
    @SerialName("dop_info") var dop_info: String? = null
    @SerialName("image_url") var image_url: String? = null

    constructor(fio_doctor: String, titleSpec : String) {
        this.fio_doctor = fio_doctor
        this.name_specialties = titleSpec
    }

    constructor(
        id: Int,
        fio_doctor: String?,
        id_specialties_string: String?,
        experience: String?,
        name_specialties: String?,
        dop_info: String?,
        image_url: String?,
    ) {                     //for ios
        this.id = id
        this.fio_doctor = fio_doctor
        this.id_specialties_string = id_specialties_string
        this.experience = experience
        this.name_specialties = name_specialties
        this.dop_info = dop_info
        this.image_url = image_url
    }

    fun  getIdSpecialtiesIntList() : MutableList<Int>?{
        return setIdSpecialtiesString(id_specialties_string)
    }
//    fun  getIdSpecialtiesStringList() : MutableList<String>?{    //for ios
//        val tmp = setIdSpecialtiesString(id_specialties_string) ?: return null
//
//        var tmpNew = mutableListOf<String>()
//        for(i in tmp){
//            tmpNew.add(i.toString() )
//        }
//        return tmpNew
//    }

    fun setIdSpecialtiesString(id_specialties_string : String?) : MutableList<Int>?
    {
        this.id_specialties_string = id_specialties_string;

        if (this.id_specialties_string == null || this.id_specialties_string!!.isEmpty())
            return null

        var list : MutableList<Int> = mutableListOf()
        val numberCommas = getNumberOfCommas (this.id_specialties_string!!) + 1;

        if (numberCommas == 1) {
            list.add(this.id_specialties_string!!.toInt());
            return list
        }

        var str =this.id_specialties_string!!
        for (i in 0 until numberCommas)
        {
            val num = str.indexOf (",");

            if (num != -1) {
                val spec = str.substring (0, num);
                list.add(spec.toInt());
                str = str.substring(num + 1, str.length);
            } else {
                try {
                    list.add(str.toInt());
                } catch (e : Exception) { }
                break;
            }
        }

        return list
    }


    fun getNumberOfCommas(str : String) : Int
    {
        var commas = 0;
        for (ch in str.toCharArray())
        {
            if (ch == ',')
                commas++
        }
        return commas
    }
}